packages/ytube/src/lib/youtubeClient.ts

```diff
@@ -1,202 +1,296 @@
-    )()
-  }
-  downloadVideo = (videoUrl: string): Readable => {
-    return ytdl(videoUrl);
-  }
-  private async iterateVideos(youtube: youtube_v3.Youtube, channel: Channel, max: number){
-    let videos: Video[] = [];
-    let continuation: string;
+    const tokenResponse = await this.getAccessToken(code)
+    const tokenInfo = await this.getAuth().getTokenInfo(tokenResponse.tokens.access_token)
+
+    const user = new User(
+      tokenInfo.sub,
+      tokenInfo.email,
+      tokenResponse.tokens.access_token,
+      tokenResponse.tokens.refresh_token,
+      code
+    )
+    return user
+  }
+
+  async getChannels(user: User) {
+    const yt = this.getYoutube(user.accessToken, user.refreshToken)
+
+    const channelResponse = await yt.channels.list({
+      part: ['snippet', 'contentDetails', 'statistics'],
+      mine: true,
+    })
+    const channels = this.mapChannels(user, channelResponse.data.items ?? [])
+
+    return channels
+  }
+
+  async verifyChannel(channel: Channel): Promise<Channel> {
+    const errors: ChannelVerificationFailed[] = []
+    if (channel.statistics.subscriberCount < MINIMUM_SUBSCRIBERS_COUNT) {
+      errors.push({
+        errorCode: ExitCodes.CHANNEL_CRITERIA_UNMET_SUBSCRIBERS,
+        message:
+          `Channel ${channel.id} with ${channel.statistics.subscriberCount} subscribers does not ` +
+          `meet Youtube Partner Program requirement of ${MINIMUM_SUBSCRIBERS_COUNT} subscribers`,
+        result: channel.statistics.subscriberCount,
+        expected: MINIMUM_SUBSCRIBERS_COUNT,
+      })
+    }
+
+    // at least MINiMUM_VIDEO_COUNT videos should be one month old
+    const oneMonthsAgo = new Date()
+    oneMonthsAgo.setMonth(oneMonthsAgo.getMonth() - MINIMUM_VIDEO_AGE_MONTHS)
+
+    // filter all videos that are older than one month
+    const videos = (await this.getVideos(channel, MINiMUM_VIDEO_COUNT)).filter(
+      (v) => new Date(v.publishedAt) < oneMonthsAgo
+    )
+    if (videos.length < MINiMUM_VIDEO_COUNT) {
+      errors.push({
+        errorCode: ExitCodes.CHANNEL_CRITERIA_UNMET_VIDEOS,
+        message:
+          `Channel ${channel.id} with ${channel.statistics.videoCount} videos does not meet Youtube ` +
+          `Partner Program requirement of at least ${MINiMUM_VIDEO_COUNT} videos, each 1 month old`,
+        result: channel.statistics.videoCount,
+        expected: MINiMUM_VIDEO_COUNT,
+      })
+    }
+
+    // Channel should be at least 3 months old
+    const threeMonthsAgo = new Date()
+    threeMonthsAgo.setMonth(threeMonthsAgo.getMonth() - MINIMUM_CHANNEL_AGE_MONTHS)
+    if (new Date(channel.publishedAt) > threeMonthsAgo) {
+      errors.push({
+        errorCode: ExitCodes.CHANNEL_CRITERIA_UNMET_CREATION_DATE,
+        message:
+          `Channel ${channel.id} with creation time of ${channel.publishedAt} does not ` +
+          `meet Youtube Partner Program requirement of channel being at least 3 months old`,
```

Comment:
`at least 3 months old` should use `MINIMUM_CHANNEL_AGE_MONTHS`

Reply:
addressed in [5c475a3](https://github.com/Joystream/youtube-synch/pull/40/commits/5c475a30e429375f666b84d4b368e5387e68ee42)